src/queryValidator.js

```diff
@@ -0,0 +1,78 @@
+import { cmcdTypes, keyTypes, errorTypes } from './constants.js';
+import { createError } from './error.js';
+import checkQuotes from './utils/checkQuotes.js';
+
+const queryValidator = (queryString, error) => {
+  let valid = true;
+
+  // Check if the URL is encoded
+  if (decodeURI(queryString) === queryString) {
+    error.push(createError(errorTypes.parameterEncoding));
+    return {
+      valid: false,
+      queryString,
+    };
+  }
+
+  // Check if there is more than one CMCD request
+  const query = queryString.split('?').pop();
+  const requests = decodeURIComponent(query).split('CMCD=');
+  requests.shift();
+
+  if (requests.length > 1) {
+    error.push(createError(errorTypes.incorrectFormat));
+    return {
+      valid: false,
+      queryString,
+    };
+  }
+
+  const values = decodeURIComponent(query).split('CMCD=')[1].split(',');
+  const [rightValue] = values[values.length - 1].split('&');
+  values[values.length - 1] = rightValue;
+
+  // console.log("values\n", values);
+  const keys = [];
+
+  // Check: key/value is separated by =
+  values.forEach((val) => {
+    const [key, value] = val.split('=');
+    keys.push(key);
+
+    // Check: string require ""
+
+    if ((keyTypes[key] === cmcdTypes.string && !checkQuotes(value))
+      || (keyTypes[key] === cmcdTypes.token && checkQuotes(value))
+    ) {
+      valid = false;
+      error.push(createError(errorTypes.invalidValue, key, value));
+    }
+
+    // Check: if the key does not have value it must be a bool
+    if (
+      (typeof value === 'undefined' && keyTypes[key] !== cmcdTypes.boolean)
+      || (value === 'true' && typeof value === 'string')
```

Comment:
I think this should be typeof value === cmcdTypes.boolean instead of typeof value === 'string'.